Modules/Devices/src/Devices.Domain/Entities/Identities/IdentityDeletionProcessAuditLogEntry.cs

```diff
@@ -191,6 +191,43 @@ public static IdentityDeletionProcessAuditLogEntry GracePeriodReminder3Sent(Iden
             DeletionProcessStatus.Approved
         );
     }
+
+    public static IdentityDeletionProcessAuditLogEntry DataDeleted(IdentityDeletionProcessId processId, IdentityAddress identityAddress, string aggregateType)
+    {
+        if (!TryGetMessageKey(aggregateType, out var messageKey))
+            throw new ArgumentException("Invalid aggregateType", nameof(aggregateType));
+
+        return new IdentityDeletionProcessAuditLogEntry(
+            processId,
+            messageKey,
+            Hasher.HashUtf8(identityAddress.Value),
+            null,
+            DeletionProcessStatus.Deleting,
+            DeletionProcessStatus.Deleting
+        );
+    }
+
+    private static bool TryGetMessageKey(string aggregateType, out MessageKey messageKey)
+    {
+        var messageKeyMappings = new Dictionary<string, MessageKey>
+        {
+            { "Challenges", MessageKey.Challenges },
+            { "PnsRegistrations", MessageKey.PnsRegistrations },
+            { "Identity", MessageKey.Identities },
```

Comment:
This should be `"Identities"` as it is written in the enum.

Reply:
Changes made.